spec/rubocop/cop/rspec/capybara/visibility_matcher_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::Capybara::VisibilityMatcher do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense when using `visible: true`' do
+    expect_offense(<<-RUBY)
+      expect(page).to have_selector('.my_element', visible: true)
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `:visible` instead of `true`.
+    RUBY
+  end
+
+  it 'registers an offense when using `visible: false`' do
+    expect_offense(<<-RUBY)
+      expect(page).to have_selector('.my_element', visible: false)
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `:all` or `:hidden` instead of `false`.
+    RUBY
+  end
+
+  it 'registers an offense when using a selector`' do
+    expect_offense(<<-RUBY)
+      expect(page).to have_selector(:css, '.my_element', visible: false)
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `:all` or `:hidden` instead of `false`.
+    RUBY
+  end
+
+  it 'registers an offense when using a using multiple options`' do
+    expect_offense(<<-RUBY)
+      expect(page).to have_selector('.my_element', count: 1, visible: false, normalize_ws: true)
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `:all` or `:hidden` instead of `false`.
```

Comment:
Instead of
```
expect(page).to have_selector('.my_element', count: 1, visible: false, normalize_ws: true)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `:all` or `:hidden` instead of `false`.
```
I would prefer
```
expect(page).to have_selector('.my_element', count: 1, visible: false, normalize_ws: true)
                                                       ^^^^^^^^^^^^^^ Use `:all` or `:hidden` instead of `false`.
```

That is, highlighting _only_ the offending part of the hash argument.

You can do this by having the node matchers return that part of the node by using `$`, e.g.
```
(send nil? :have_selector ... (hash <$(pair (sym :visible) true) ...>))
```
and then using the return value (which is also yielded, if you call the node matcher with a block) – e.g.
```rb
            visible_true?(node) do |arg|
              add_offense(arg, message: MSG_TRUE)
            end
```

Reply:
Good point, I didn't know how to do that, thanks for the pointers. I've added these changes and am now waiting for CI. I will squash if you approve the changes.

Reply:
Go ahead, looks perfect.

Reply:
Done.